tensorpipe/channel/cuda_xth/context_impl.cc

```diff
@@ -62,7 +63,28 @@ std::shared_ptr<CudaChannel> ContextImpl::createChannel(
 }
 
 bool ContextImpl::isViable() const {
-  return foundCudaLib_;
+  if (!foundCudaLib_) {
+    return false;
+  }
+
+  int deviceCount;
+  TP_CUDA_CHECK(cudaGetDeviceCount(&deviceCount));
+  for (int i = 0; i < deviceCount; ++i) {
+    cudaDeviceProp props;
+    TP_CUDA_CHECK(cudaGetDeviceProperties(&props, i));
+
+    // Unified addressing is required for cross-device `cudaMemcpyAsync()`. We
+    // could lift this requirement by adding a fallback to
+    // `cudaMemcpyPeerAsync()`.
```

Comment:
If you have a reference it would be nice to link to it here. Otherwise it's fine too, unified addressing seems common enough that there's no real need to justify it.